src/functions-and-arrays.js

```diff
@@ -1,86 +1,200 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 >= num2) {
+    return num1;
+  } else {
+    return num2;
+  }
+}
 
+// Iteration #2: Find longest word
+const words = ["mystery", "brother", "aviator", "crocodile", "pearl", "orchard", "crackpot"];
 
+function findLongestWord(words) {
+  if (words.length === 0) {
+    return null;
+  }
 
-// Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+  if (words.length === 1) {
+    return words[0];
+  }
 
-function findLongestWord() {}
+  let wordLengths = words.map((word) => {
+    return word.length;
+  });
 
+  let longestWordLength = Math.max(...wordLengths);
 
+  let longestWords = words.filter((word) => {
+    return word.length === longestWordLength;
+  });
+
+  return longestWords[0];
+}
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
+function sumNumbers(numbers) {
+  let sum = numbers.reduce((acc, num) => {
+    return acc + num;
+  }, 0);
 
+  return sum;
+}
 
 // Iteration #3.1 Bonus:
-function sum() {}
-
-
+function sum(mixedArray) {
+  let uniqueObjects = [
+    ...new Set(
+      mixedArray.map((item) => {
+        return typeof item;
+      })
+    ),
+  ];
+
+  allowedObjects = ["number", "string", "boolean"];
+
+  let nonAllowedObjects = uniqueObjects.filter((item) => {
+    return item !== "number" && item !== "string" && item !== "boolean";
+  });
+
+  if (nonAllowedObjects.length > 0) {
+    throw new Error("Unsupported data type sir or ma'am");
+  }
+
+  let sumNum = mixedArray
```

Comment:
Also here it would be more efficient to have one loop with multiple conditions than multiple loops